src/infill/ZigzagConnectorProcessor.h

```diff
@@ -94,59 +80,130 @@ namespace cura
+    void registerScanlineSegmentIntersection(const Point& intersection, int scanline_index, int direction);
+
     /*!
      * Handle the end of a polygon and prepare for the next.
      * This function should reset all member variables.
      */
-    virtual void registerPolyFinished() = 0;
+    void registerPolyFinished();
+
+    /*!
+     * Reset the state so it can be used for processing another polygon.
+     */
+    void reset();
+
+protected:
+    const PointMatrix& rotation_matrix; //!< The rotation matrix used to enforce the infill angle
+    Polygons& result; //!< The result of the computation
+
+    bool use_endpieces; //!< Whether to include end pieces or not
+    bool connected_endpieces; //!< Whether the end pieces should be connected with the rest part of the infill
+    int skip_some_zags; //!< Whether to skip some zags
+    int zag_skip_count; //!< Skip 1 zag in every N zags
+
+    bool is_first_connector; //!< indicating whether we are still looking for the first connector or not
+    int first_connector_index; //!< scanline segment index of the first connector
+    int first_connector_direction; //!< direction of the first connector
+    int last_connector_index; //!< scanline segment index of the last connector
+    int last_connector_direction; //!< direction of the last connector
+
+    /*!
+     * The line segments belonging the zigzag connector to which the very first vertex belongs. 
+     * This will be combined with the last handled zigzag_connector, which combine to a whole zigzag connector.
+     * 
+     * Because the boundary polygon may start in in the middle of a zigzag connector, 
+     */
+    std::vector<Point> first_connector; 
+    /*!
+     * The currently built up zigzag connector (not the first/last) or end piece or discarded boundary segment
+     */
+    std::vector<Point> current_connector; 
+
+    /*!
+     * Add a line to the result bu unapplying the rotation rotation_matrix.
+     * 
+     * \param from The one end of the line segment
+     * \param to The other end of the line segment
+     */
+    void addLine(Point from, Point to);
+
+    /*!
+     * Checks whether this connector should be added or not.
+     *
+     * \param start_scanline_idx the start scanline index of this scanline segment
+     * \param end_scanline_idx The the end scanline index of this scanline segment
+     * \param direction The direction of this scanline segment
+     */
+    bool shouldAddThisConnector(int start_scanline_idx, int end_scanline_idx, int direction) const;
 };
 
+//
+// Inline functions
+//
+
+inline void ZigzagConnectorProcessor::reset()
+{
+    this->is_first_connector = true;
+    this->first_connector_index = 0;
+    this->first_connector_direction = 0;
+    this->last_connector_index = 0;
+    this->last_connector_direction = 0;
+    this->first_connector.clear();
+    this->current_connector.clear();
+}
+
+inline void ZigzagConnectorProcessor::addLine(Point from, Point to)
+{
+    assert(from != to);
```

Comment:
I think it is caused by a small mistake... 4e04807818939470e0bb0974fc7ea79d3502d056